cogs/dank_memes.py

```diff
@@ -157,3 +248,42 @@ class DankMemes(BaseCog):
                 await ctx.send(index, file=discord.File(os.path.join(MEDIA_PATH, images[index])))
             except (ValueError, RuntimeError, IndexError):
                 await ctx.send('That is not a valid meme')
+
+    @commands.command()
+    async def deepfry(self, ctx, index: typing.Optional[int]):
+        """
+        Deep frys an image
+
+        Parameters
+        ----------
+        ctx : discord.ext.commands.Context
+        index : int (optional)
+
+        Returns
+        -------
+        Image
+            Deep Fried Image
+        """
+        if ctx.message.attachments:
+            async with ctx.message.channel.typing():
+                for attachment in ctx.message.attachments:
+                    response = requests.get(attachment.url)
+                    if response.status_code == 200:
+                        try:
+                            file = self._deep_fry(Image.open(BytesIO(response.content)))
+                        except OSError:
+                            await ctx.message.channel.send('I can\'t deep fry that file')
+                        else:
+                            await ctx.message.channel.send(file=discord.File(file.name))
+                    else:
+                        await ctx.message.channel.send('I could not download your file')
+        elif index is not None:
+            try:
+                if index < 0:
+                    raise IOError
+                file = self._deep_fry(Image.open(MEDIA_PATH + '{}.jpg'.format(index)))
```

Comment:
If we're going to incorporate both this change and #21, then we should switch all checks to `index` to `index + 1` to  fit with the changes being introduced in #21